packages/apps/sale/src/components/organisms/DropDownSaleOrderView/DropDownSaleOrderView.tsx

```diff
@@ -0,0 +1,190 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2024 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, {useCallback, useEffect, useMemo} from 'react';
+import {StyleSheet, TouchableOpacity, View} from 'react-native';
+import {Icon, Text, useThemeColor} from '@axelor/aos-mobile-ui';
+import {
+  isEmpty,
+  useNavigation,
+  useSelector,
+  useTranslator,
+  getLastItem,
+  useTypes,
+  useDispatch,
+  useIsFocused,
+} from '@axelor/aos-mobile-core';
+import {SaleOrderCard} from '../../atoms';
+import {fetchSaleOrder} from '../../../features/saleOrderSlice';
+
+const DropDownSaleOrderView = ({customer}) => {
+  const I18n = useTranslator();
+  const navigation = useNavigation();
+  const dispatch = useDispatch();
+  const Colors = useThemeColor();
+  const isFocused = useIsFocused();
+
+  const {SaleOrder} = useTypes();
+
+  const {saleOrderList} = useSelector(state => state.sale_saleOrder);
+
+  useEffect(() => {
+    if (isFocused) {
+      dispatch((fetchSaleOrder as any)({customerId: customer?.id}));
+    }
+  }, [customer, dispatch, isFocused]);
+
+  const _saleOderList = useMemo(() => {
+    return saleOrderList.filter(
+      saleOrder =>
+        saleOrder?.statusSelect === SaleOrder?.statusSelect.Confirmed ||
+        saleOrder?.statusSelect === SaleOrder?.statusSelect.Completed,
+    );
+  }, [
+    SaleOrder?.statusSelect.Completed,
+    SaleOrder?.statusSelect.Confirmed,
+    saleOrderList,
+  ]);
+
+  const quotationList = useMemo(() => {
+    return saleOrderList.filter(
+      saleOrder =>
+        saleOrder?.statusSelect === SaleOrder?.statusSelect.Draft ||
+        saleOrder?.statusSelect === SaleOrder?.statusSelect.Finalized,
+    );
+  }, [
+    SaleOrder?.statusSelect.Draft,
+    SaleOrder?.statusSelect.Finalized,
+    saleOrderList,
+  ]);
+
+  const lastOrderList = useMemo(() => {
+    return getLastItem(_saleOderList, 'creationDate');
+  }, [_saleOderList]);
+
+  const lasQuotationList = useMemo(() => {
+    return getLastItem(quotationList, 'creationDate');
+  }, [quotationList]);
```

Comment:
Names are not right as this is not a list but only an item : 
* lastOrder
* lastQuotation